### PR TITLE
[CMake] Fold export_executable_symbols_* into function args.

### PR DESCRIPTION
**Pull request by cachemeifyoucan**
`LLVM_ENABLE_EXPORTED_SYMBOLS_IN_EXECUTABLES` is not completely
compatible with `export_executable_symbols` as the later will be ignored
if the previous is set to NO.

 Fix the issue by passing if symbols need to be exported to
`llvm_add_exectuable` so the link flag can be determined directly
without calling `export_executable_symbols_*` later.

### CODE REVIEW

**Review by cyndyishida — APPROVED** [review]
This only changes exports for `llvm-jitlink` and not any of the other toolchain executables, LGTM! Thank you Steven!

**Comment by cachemeifyoucan** [comment]
> This only changes exports for `llvm-jitlink` and not any of the other toolchain executables, LGTM! Thank you Steven!

Yes. There are few other executable that uses `export_executable_symbols`. It would be good to audit them to see if they need to either just don't export, or apply the same workaround. I don't see them failing in CI so I left them out in the PR.

File: llvm/examples/Kaleidoscope/BuildingAJIT/Chapter1/CMakeLists.txt [hunk 1]
```diff
@@ -14,5 +14,3 @@ set(LLVM_LINK_COMPONENTS
 add_kaleidoscope_chapter(BuildingAJIT-Ch1
   toy.cpp
   )
-
-export_executable_symbols(BuildingAJIT-Ch1)
```
Comment (lhames) [inline llvm/examples/Kaleidoscope/BuildingAJIT/Chapter1/CMakeLists.txt:18]: All tutorial exercises and examples that currently use `export_executable_symbols` will now need `EXPORT_SYMBOLS`. The example/tutorial tests are off by default and Darwin exports symbols by default so we may not see any immediate bot failures, but the tests will fail on Linux where symbols are not exported by default.

**Review by lhames — CHANGES_REQUESTED** [review]

File: llvm/examples/Kaleidoscope/BuildingAJIT/Chapter1/CMakeLists.txt (hunk 1 above)
Comment (cachemeifyoucan) [reply llvm/examples/Kaleidoscope/BuildingAJIT/Chapter1/CMakeLists.txt:18]: The expert symbols are pushed to the add_llvm_example by default

Comment (lhames) [reply llvm/examples/Kaleidoscope/BuildingAJIT/Chapter1/CMakeLists.txt:18]: > The expert symbols are pushed to the add_llvm_example by default

Totally missed that. Thanks for the clarification!

**Review by lhames — APPROVED** [review]

**Comment by petrhosek** [comment]
This breaks the build with the following error:
```
CMake Error at cmake/modules/AddLLVM.cmake:970 (add_library):
  Cannot find source file:

    EXPORT_SYMBOLS_FOR_PLUGINS
Call Stack (most recent call first):
  /b/s/w/ir/x/w/llvm-llvm-project/lld/cmake/modules/AddLLD.cmake:48 (generate_llvm_objects)
  /b/s/w/ir/x/w/llvm-llvm-project/lld/tools/lld/CMakeLists.txt:6 (add_lld_tool)


CMake Error at cmake/modules/AddLLVM.cmake:970 (add_library):
  No SOURCES given to target: obj.lld
Call Stack (most recent call first):
  /b/s/w/ir/x/w/llvm-llvm-project/lld/cmake/modules/AddLLD.cmake:48 (generate_llvm_objects)
  /b/s/w/ir/x/w/llvm-llvm-project/lld/tools/lld/CMakeLists.txt:6 (add_lld_tool)
```
Would it be possible to revert the change?

**Comment by cachemeifyoucan** [comment]
Yes, please revert. Do you have link to the build failure?

**Comment by cachemeifyoucan** [comment]
Reverted. I suspect it is a LLVM_DRIVER build that is not covered by PR testing and I missed some code path there. Let me improve later.

**Pull request merged**